fix: Fix pascal enum reference

sschw opened the pull request:
Generating enums using the useEnums option will generate the enum types with `pascal(name)`, however when referencing the enum type, the name is not formatted.

This causes issues with lowerCamelCase names.

Wrapping `name || ""` inside a `pascal` solves issues with lowerCamelCase names.

fabien0102 commented:
Can you add a unit test to demonstrate the issue? I'm not sure I see what you are fixing 🤔

sschw commented:
Sure, I added a test to schemaToTypeAliasDeclaration.test.ts.
It can easily be tested by changing `"Test"` to `"test"` in the printSchema function in schemaToEnumDeclaration.test.ts and schemaToTypeAliasDeclaration.test.ts.
While schemaToEnumDeclaration is still working as intended, the enum tests in schemaToTypeAliasDeclaration was failing.

I also added a test to schemaToEnumDeclaration.test.ts, just so that this case is tested there as well.

sschw commented:
@fabien0102 Rebased this PR and solved the conflicts as jest update changed the snapshots.

fabien0102 reviewed (approved):
👌
